contracts/vaults/common/StrategyUtils.sol

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity 0.8.17;
+
+import {IERC20} from "../../../interfaces/IERC20.sol";
+import {
+    TradeParams,
+    DepositTradeParams,
+    RedeemParams,
+    SingleSidedRewardTradeParams
+} from "../../../interfaces/notional/ISingleSidedLPStrategyVault.sol";
+import {TradeHandler} from "../../trading/TradeHandler.sol";
+import {Deployments} from "../../global/Deployments.sol";
+import {Constants} from "../../global/Constants.sol";
+import {Errors} from "../../global/Errors.sol";
+import {ITradingModule, Trade, TradeType, DexId} from "../../../interfaces/trading/ITradingModule.sol";
+
+/**
+ * @notice External library deployed for the purposes of handling SingleSidedLP trades. All
+ * the methods in this library are called inside a `delegateCall` context which ensures that
+ * the library has access to the calling vault's token balances
+ */
+library StrategyUtils {
+    using TradeHandler for Trade;
+
+    /// @notice Trades the amount of primary token into other secondary tokens prior
+    /// to entering a pool.
+    function executeDepositTrades(
+        IERC20[] memory tokens,
+        uint256[] memory amounts,
+        DepositTradeParams[] memory depositTrades,
+        uint256 primaryIndex
+    ) external returns (uint256[] memory) {
+        address primaryToken = address(tokens[primaryIndex]);
+
+        for (uint256 i; i < amounts.length; i++) {
+            if (i == primaryIndex) continue;
+            DepositTradeParams memory t = depositTrades[i];
+            // Do not allow ZERO_EX trading in this method since we cannot validate
+            // the arbitrary exchange data.
+            if (DexId(t.tradeParams.dexId) == DexId.ZERO_EX) revert Errors.InvalidDexId(uint256(DexId.ZERO_EX));
+
+            if (t.tradeAmount > 0) {
+                // Always selling the primaryToken and buying the secondary token.
+                (uint256 amountSold, uint256 amountBought) = _executeDynamicSlippageTradeExactIn(
+                    Deployments.TRADING_MODULE, t.tradeParams, primaryToken, address(tokens[i]), t.tradeAmount
+                );
+
+                amounts[i] = amountBought;
+                // Will revert on underflow if over-selling the primary borrowed
+                amounts[primaryIndex] -= amountSold;
+            }
+        }
+
+        return amounts;
+    }
+
+    /// @notice Trades the amount of secondary tokens into the primary token after
+    /// exiting a pool.
+    function executeRedemptionTrades(
+        IERC20[] memory tokens,
+        uint256[] memory exitBalances,
+        TradeParams[] memory redemptionTrades,
+        uint256 primaryIndex
+    ) external returns (uint256 finalPrimaryBalance) {
+        address primaryToken = address(tokens[primaryIndex]);
+
+        for (uint256 i; i < exitBalances.length; i++) {
+            if (i == primaryIndex) finalPrimaryBalance += exitBalances[i];
```

Comment:
Shouldn't you add a continue statement here after incrementing finalPrimaryBalance?